### ISSUE TITLE
My location GPS button in UI

### ISSUE THREAD

**Issue by sharmaishant92** [keita-makino/qualtrics-map#43]
Hi @keita-makino ,
I have a query. Is it possible to add my location button in the UI so that participant does not have to scroll to find their location (as we already know that the less complexity in the survey, better it is)? 
This way participants will be able to start automatically from their home location and select the point. 
Please let me know if it is doable with an easy fix or code replacement in bundle script.

**Comment by keita-makino** [comment]
Hi @sharmaishant92 sorry I overlooked the inquiry.

That feature was indeed the default option (i.e., participants would see their location on the map upon entering the question and click "Allow GPS" option on the browser) but omitted due to:

1. Privacy concern
People (especially those who are not familiar with IT standards) may get anxious about the quality of the survey as they feel they are tracked by us.
2. Burden
It requires just one more click and consent during the survey which could be annoying for the participants. I'm not so informed about the default option of the Qualtrics surveys, but I believe the platform collects the country code and zip code (e.g., US-95616) that you can use in the survey. In my previous project I was using those attributes but maybe it didn't work in your case, according to your past inquiry for the default location?

In short, though, I can implement it if you (and some others in future) really need it.

**Comment by sharmaishant92** [comment]
Hi @keita-makino . Thanks for the response. 
I do agree with the privacy concern, and it will add extra consent as per IRB rules, and excluding it makes sense. Because the area we want to survey is pretty spread, it will be a challenge or frustrating for the participants to scroll to the location we want to ask. 
To relax that further, I have one additional query. Is it possible to start the initial position of the map from a 5-digit zip code captured through a response to the previous question without asking the participants to allow location access?

**Comment by keita-makino** [comment]
Hi @sharmaishant92, I think it is possible to use the previous question, but it will require a decent work on JavaScript code (if Qualtrics does not provide a convenient way to retrieve a response to one of the previous questions).

In the first place, the current version can let users first input their zipcode, click one of the suggestions and then fly to the location so that they can further drag and drop the pin, Wouldn't this feature work?

**Comment by keita-makino** [comment]
Closed with inactivity.
